test/apps/index.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const assert = require('assert');
+
+describe('apps', function () {
+	const libApp = require('lib/app');
+
+	// allow lib/app configuration override
+
+	let currentAppsConfig;
+	let currentCommandCenterSetup;
+
+	const oldFns = {
+		getAppsConfig: libApp.getAppsConfig,
+		getLogger: libApp.getLogger,
+		createCommandCenter: libApp.createCommandCenter
+	};
+
+	const testFns = {
+		getAppsConfig: function () {
+			return currentAppsConfig;
+		},
+		getLogger: function () {
+			return {
+				debug: function () {}
+			};
+		},
+		createCommandCenter: function (app) {
+			return {
+				setup: function () {
+					currentCommandCenterSetup(app);
+				}
+			};
+		}
+	};
+
+	beforeEach(function () {
+		libApp.getAppsConfig = testFns.getAppsConfig;
```

Comment:
Maybe not something for right now, but food for thought. I've found [sinon](http://sinonjs.org/) extremely useful for spying and stubbing. It avoids a lot of boiler plate and even helps me to formulate tests. `oldFns` and `testFns` above can be avoided by doing something like:

```javascript
const sinon = require('sinon');
const sandbox = sinon.sandbox.create();

before(() => {
    sandbox.stub(libApp, 'getAppsConfig').returns(currentAddsConfig);
    sandbox.stub(libApp, 'getLogger').returns(() => {
        return { debug: sandbox.stub() };
    }));
    sandbox.stub(libApp, 'createCommandCenter').returns(app => {
        return {
            setup() {
                currentCommandCenterSetup(app);
            }
        };
    }));
});

afterEach(() => {
    sandbox.reset(); // Reset stub call counts, args, etc.
    libApp.removeAllApps();
});

after(() => {
    sandbox.restore(); // Unstub.
});
```

The above is more like a mock, so you don't really care about inputs or call order. Stubs do record those though, which can be really handy. The sandbox here isn't strictly necessary, but it helps to ensure that stubbed methods aren't forgotten and leak into other tests.